fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java

```diff
@@ -267,19 +268,21 @@ protected CompletableFuture<Tuple> unsupportedAggregateFunction(@Nonnull IndexAg
     public abstract CompletableFuture<IndexOperationResult> performOperation(@Nonnull IndexOperation operation);
 
     /**
-     * Use the index prefetch scan method to return a range scan with the dereferenced record for each index entry.
-     * @param range the index range to scan
-     * @param mapper the mapper construct that is used to build a record primary key from an index entry
+     * Use the getMappedRange scan method to return a range scan with the dereferenced record for each index entry.
+     * @param scanBounds the index range to scan
      * @param continuation the continuation to use
      * @param scanProperties the scan properties to use
+     * @param recordSubspace the record subspace: The prefix of the keyspace to use for the derefenced records
+     * @param commonPrimaryKey the common primary key for the derefenced records
      * @return a cursor of the index prefetch call result: will contain both index entries and dereferenced records
      */
     @Nonnull
-    public RecordCursor<FDBIndexedRawRecord> scanIndexPrefetch(@Nonnull final TupleRange range,
-                                                               @Nonnull final byte[] mapper,
-                                                               @Nullable final byte[] continuation,
-                                                               @Nonnull final ScanProperties scanProperties) {
+    public RecordCursor<FDBIndexedRawRecord> scanRemoteFetch(@Nonnull final IndexScanBounds scanBounds,
+                                                             @Nullable final byte[] continuation,
+                                                             @Nonnull final ScanProperties scanProperties,
+                                                             @Nonnull final Subspace recordSubspace,
```

Comment:
I think the `recordSubspace` parameter shouldn't be needed here, because the index maintainer can call `state.store.recordsSubspace()` to get the same thing

Reply:
That felt a little too restrictive, as it would create a situation where the call relies on the store to set things up and then assumes that the record subspace is taken from the store within the index maintainer. It seemed better separation and testability for the record concerns are given to the method in a formal parameter. That said, I don't feel strongly about it, so will change if you think it better.

Reply:
I'm not sure I follow. The reason I suggested relying on the `state.store.recordsSubspce()` is that an index maintainer should never be told to scan from any records subspace except for the one associated with the store from which it's created, so my concern with having `recordSubspace` as a parameter here is mainly that someone might accidentally call `scanRemoteFetch` and pass in the wrong store's subspace.

But I'm not sure I follow the problem you're seeing regarding the store setting things up and then relying on store subspace being the same (unless the thing you're talking about setting up is the `mapper` parameter, which is also pushed down into the maintainer now).

Reply:
Sure. I misunderstood the contract between the classes and so my assumption is invalidated. Since the contract is that the record subspace is provided to the `IndexMaintainer` by the store as is a security measure, it makes sense not to provide that parameter externally. Changed as suggested.